### PR TITLE
Homework5

### PR DESCRIPTION
**Pull request by Tryxman**
Did this homework before watching the part of the lecture where this homework is explained

### CODE REVIEW

File: HomeTask5.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import Darwin
+
+let wholeNumberArray = [1, 2, 3, 4, 5, 6, 10, 14, 16, 19, 21, 34]
+
+let evenNumbersArray = wholeNumberArray.filter({ $0 % 2 == 0 })
+let oddNumbersArray  = wholeNumberArray.filter({ $0 % 2 != 0 })
+
+//2. Дан массив опциональных строк. Посчитайте, сколько букв каждая буква встречается во всех словах.
+
+let optionalStringsArray        = ["A", "Ба", "Ба", "Дельфин", nil]
```
Comment (darya-gurinovich) [inline HomeTask5.playground/Contents.swift]: You need to put `=` either above other `=` or right after `optionalStringsArray` with one space

File: HomeTask5.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+var duplicatedSymbolsDictionaryIfLet = [String.Element : Int]()
+var duplicatedSymbolsDictionaryGuard = [String.Element : Int]()
```
Comment (darya-gurinovich) [inline HomeTask5.playground/Contents.swift]: Please, remove a space before `:`

File: HomeTask5.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+// Because of Numeric protocol limitation , for division using -> FloatingPoint
+func divideTwoValues<T: FloatingPoint>(_ firstNumber: T?,  by secondNumber: T?) -> T? {
+    return (firstNumber ?? 0) / (secondNumber ?? 0)
+}
+
+/*4.* Создайте функцию, которая проверяет является ли переданная строка палиндромом. Строка может быть как одним словом (mom) так и предложением (Too bad I hid a boot). В предложении вы должны игнорировать пробелы, большие или маленькие это буквы и знаки препинания при определении полиндрома.*/
+
+let originalString      = ("Mama , : -- , -  a Amam")
+let symbolsToExlclude   = ["!", " ", ",", ":", "?", "/", "(", ")", "-", "_", "+", "="]
```
Comment (darya-gurinovich) [inline HomeTask5.playground/Contents.swift]: You've misspelled `symbolsToExclude `

File: HomeTask5.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+// Because of Numeric protocol limitation , for division using -> FloatingPoint
+func divideTwoValues<T: FloatingPoint>(_ firstNumber: T?,  by secondNumber: T?) -> T? {
+    return (firstNumber ?? 0) / (secondNumber ?? 0)
+}
+
+/*4.* Создайте функцию, которая проверяет является ли переданная строка палиндромом. Строка может быть как одним словом (mom) так и предложением (Too bad I hid a boot). В предложении вы должны игнорировать пробелы, большие или маленькие это буквы и знаки препинания при определении полиндрома.*/
+
+let originalString      = ("Mama , : -- , -  a Amam")
```
Comment (darya-gurinovich) [inline HomeTask5.playground/Contents.swift]: You can drop `()` here. They are absolute in this place

File: HomeTask5.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+
+var elimnetedMethod = [String]()
+var newMethod  = ["a", "b", "c", "d", "e", "f", "g", "h", "i"]
+
+for i in newMethod {
+    elimnetedMethod.append(i)
+}
+
+print(elimnetedMethod)
```
Comment (darya-gurinovich) [inline HomeTask5.playground/Contents.swift]: What is this code? I believe `elimnetedMethod` array will be the same as `newMethod` array

**Review by darya-gurinovich — APPROVED** [review]
Great work! Well done! Good style and naming. Just a couple of things that can be improved

File: HomeTask5.playground/Contents.swift (hunk 5 above)
Comment (Tryxman) [reply HomeTask5.playground/Contents.swift]: Oops , i'm sorry , i forgot to delete this part of code )
It is my training part , brainstorm code )

**Pull request merged**